bindings/go/oci/internal/lister/component/component_versions.go

```diff
@@ -91,10 +93,13 @@ func ReferenceTagVersionResolver(component string, store interface {
 				return "", fmt.Errorf("failed to decode index for tag %q: %w", tag, err)
 			}
```

Comment:
By this time, after fetch has returned with a `data` but no error, but the decoding isn't working, you effectively lock the `store` in data, that you aren't freeing up in the NewDecoder's error now. So now, you need to add `data.Close` in there error cases as well if I understand the locking mechanism correctly.